### DISCUSSION TITLE
How to stop mpv from triggering VRR?

### DISCUSSION THREAD

**Discussion by rldlive** [mpv-player/mpv#14597] [Q&A]
I have an Nvidia GPU, 144 hz Adaptive-Sync monitor and have G-Sync Compatible mode enabled (fullscreen only!) in Nvidia Control Panel.
No matter how I open any video windowed or fullscreen it always triggers VRR and monitor reports huge refresh rate fluctuations. I can forcibly disable VRR for mpv.exe only in NVCP but then the monitor turns off and on every time I open and close mpv which is annoying as hell.
When I watch video in browsers it's always stable 144 hz with VRR enabled system wide and the monitor doesn't turn off and on which is what I expect from mpv.
I am using Windows 11 and the latest shinchiro mpv build.

**Comment by yeezylife** [comment]
> I can forcibly disable VRR for mpv.exe only in NVCP but then the monitor turns off and on every time I open and close mpv which is annoying as hell.

same situation here.

**Reply by Raza-0007** [reply]
You have multiple options to get around this issue. Following assumes that you are on Windows OS with Nvidia GPU on default settings, with g-sync/adaptive sync turned on.

- Don't use `video-sync=display-resample`. This is what triggers the issue with g-sync/adaptive sync enabled monitors.

- If you have to use `video-sync=display-resample`, then don't use `gpu-api=vulkan`. Use the default `gpu-api=d3d11`. For some reason `gpu-api=vulkan` triggers refresh rate fluctuations, but d3d11 does not. It is either because of how these API's are coded in mpv or just a difference between the API's themselves that results in d3d11 API to disable g-sync/adaptive sync during mpv playback.

- Keep g-sync/adaptive sync off and only turn them on when playing a video game.

**Reply by yeezylife** [reply]
thanks for replying. I found out that `video-sync=display-resample`+`d3d11-sync-interval=0`+ limit the max fps in nvidia control panel in 60 fps(same as monitor's refresh rate) can work with g-sync/adaptive sync without growing mistimed/delayed frames.

turning g-sync/adaptive sync on and off separately for games and mpv will cause " monitor turns off and on every time I open and close mpv or games". so I'm going with the solution above.thanks again

**Reply by Raza-0007** [reply]
>  I found out that video-sync=display-resample+d3d11-sync-interval=0+ limit the max fps in nvidia control panel in 60 fps(same as monitor's refresh rate) can work with g-sync/adaptive sync without growing mistimed/delayed frames.

I am glad that it worked for you. But setting `d3d11-sync-interval=0` disables `vsync`. I read somewhere that in order for `display-resample` to function correctly, `vsync `needs to be "ON". Otherwise the two settings will work against each other and result in choppy playback. But I may be wrong about this. Hopefully somebody more familiar with mpv's internal working can chime in and clarify. 

For me, just choosing `gpu-api=d3d11` resolves `g-sync` and `display-resample` conflict.

**Reply by yeezylife** [reply]
> > I found out that video-sync=display-resample+d3d11-sync-interval=0+ limit the max fps in nvidia control panel in 60 fps(same as monitor's refresh rate) can work with g-sync/adaptive sync without growing mistimed/delayed frames.
> 
> I am glad that it worked for you. But setting `d3d11-sync-interval=0` disables `vsync`. I read somewhere that in order for `display-resample` to function correctly, `vsync `needs to be "ON". Otherwise the two settings will work against each other and result in choppy playback. But I may be wrong about this. Hopefully somebody more familiar with mpv's internal working can chime in and clarify.
> 
> For me, just choosing `gpu-api=d3d11` resolves `g-sync` and `display-resample` conflict.

maybe I need more test. but I get growing mistimed/delayed frames again without d3d11-sync-interval=0. and i always use gpu-api=d3d11

d3d11-sync-interval=0. off:
<img width="561" height="141" alt="image" src="https://github.com/user-attachments/assets/ff8ad6e0-9edb-422d-a1f3-77293a4404b3" />
d3d11-sync-interval=0.on:
<img width="748" height="223" alt="image" src="https://github.com/user-attachments/assets/8a1c87a7-0d94-497f-a55b-b6715a3bff57" />

**Reply by Raza-0007** [reply]
Mistimed, delayed and dropped frames can sometimes be caused when using certain external shaders or scalers with `display-resample`. I noticed that when I was testing some of them a while back. I do not use any external shaders or scalers.

If your playback is smooth and you are not noticing any issues, then stick with your settings. As I mentioned before I might be wrong about `vsync` and `display-resample`. Also the info in the manual about `d3d11-sync-interval=0` [disabling vsync](https://mpv.io/manual/master/#options-d3d11-sync-interval) might not apply in `display-resample` mode; the manual is not very detailed.

**Comment by meidoqt** [comment]
You can toggle off VRR for mpv.exe with NVPI-R

<img width="858" height="704" alt="Screenshot 2025-11-07 203359" src="https://github.com/user-attachments/assets/f67ff49a-bd73-43d4-aa56-1c6c312e2aca" />
